CollapseLauncher/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,364 @@
+
+                _ = PluginInstances.TryAdd(pluginDirName, pluginInfo);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogWriteLine($"[PluginManager] Failed to load plugin from file: {pluginFile.FullName} due to error: {ex}", LogType.Error, true);
+                pluginInfo?.Dispose();
+            }
+            finally
+            {
+                if (pluginInfo != null)
+                {
+                    foreach (var currentConfig in pluginInfo.PresetConfigs)
+                    {
+                        string gameName = currentConfig.GameName;
+                        string gameRegion = currentConfig.ZoneName;
+
+                        ref Dictionary<string, PresetConfig>? dict = ref CollectionsMarshal.GetValueRefOrAddDefault(launcherMetadataConfig, gameName, out _);
+                        if (Unsafe.IsNullRef(ref dict) || dict == null)
+                        {
+                            dict = new Dictionary<string, PresetConfig>(StringComparer.OrdinalIgnoreCase);
+                        }
+
+                        dict.TryAdd(gameRegion, currentConfig);
+
+                        ref List<string>? gameRegions = ref CollectionsMarshal.GetValueRefOrAddDefault(launcherGameNameRegionCollection, gameName, out _);
+                        if (Unsafe.IsNullRef(ref gameRegions) || gameRegions == null)
+                        {
+                            gameRegions = [];
+                        }
+
+                        gameRegions.Add(gameRegion);
+
+                        long stampTimestamp = long.Parse(pluginInfo.CreationDate?.ToString("yyyyMMddHHmmss") ?? "0");
+                        _ = launcherMetadataStampDictionary.TryAdd($"{gameName} - {gameRegion}", new Stamp
+                        {
+                            GameName = gameName,
+                            GameRegion = gameRegion,
+                            LastModifiedTimeUtc = pluginInfo.CreationDate ?? DateTime.MinValue,
+                            LastUpdated = stampTimestamp,
+                            MetadataType = MetadataType.PresetConfigPlugin,
+                            PresetConfigVersion = pluginInfo.StandardVersion.ToString()
+                        });
+                    }
+                }
+            }
+        }
+    }
+
+    internal static void UnloadPlugins()
+    {
+        // Dispose all plugin instances before freeing the plugin handles.
+        foreach (var pluginInfo in PluginInstances.Values)
+        {
+            pluginInfo.Dispose();
+        }
+
+        // Clear the plugin instances.
+        PluginInstances.Clear();
+    }
+
+    private static bool IsValidReturnCode(SelfUpdateReturnCode retCode)
+    {
+        uint asUint = (uint)retCode;
+        if (retCode.HasFlag(SelfUpdateReturnCode.Ok) &&
+            asUint is >= 0b_00000001_00000000_00000000_00000000 and <= 0b_10000000_00000000_00000000_00000000)
```

Comment:
I think it would be better if the Uint is set as variable instead with clear naming